records/state/state.go

```diff
@@ -162,7 +162,8 @@ func statusIPs(st []Status, src func(*Status) []string) []string {
 	// https://github.com/apache/mesos/blob/0.24.0/src/slave/slave.cpp#L5226-L5238
 	ts, j := -1.0, -1
 	for i := range st {
-		if st[i].State == "TASK_RUNNING" && st[i].Timestamp > ts {
+		state := st[i].State
+		if (state == "TASK_RUNNING" || state == "TASK_STAGING") && st[i].Timestamp > ts {
```

Comment:
This sounds like the right way to go.  Perhaps Frameworks should provide a prioritized list of states.  In the HDFS case it would be [RUNNING, STAGING].  Meaning provide the latest RUNNING TASK if available, or fallback to STAGING, or provide nothing.

In any case, this more nuanced scheme is way outside the scope of this PR.  I personally don't have the cycles to implement and test such a scheme.  There's definitely more design work to be done too.  Did you want me to pull the discussion from here and put it in a proposal issue or something?